dts/bindings/timer/atmel,sam-tc.yaml

```diff
@@ -55,6 +55,13 @@ properties:
         the driver to count events generated on the TIOA, TIOB signal connected
         to the external pin.
 
+    reg-rc:
+      type: int
+      required: false
+      description: |
+        Initial value for the rc register. The function of this depends on how
+        the CMR register is configured.
+
```

Comment:
Thanks for the response.

I have tested this on same70 (same70 eXplained devkit) and sams70 (custom board, there are no devkit's for this part. I have a different [PR](https://github.com/zephyrproject-rtos/zephyr/pull/41576) to support this soc). The samx7x are all the same, at least regarding the timers, they all share a datasheet.

This is  pretty bare-bones driver, as it is the end user needs to be familiar with the datasheet to even know what to put in the reg-cmr register. The reg RC is only used if configured to be used via the reg-cmr, normally it is not used.

In our use case I am configuring the timer to generate an internal event when the accumulator reaches the value specified in RC. These events can be used for many things but we are using it to continuously trigger ADC conversions at a set frequency. This will be used on two different boards one with an samv70 and one with a sams70.

I have no experience with the sam3x and sam4x and unfortunately I do not have the time to learn about them right now. If that is a show stopper let me know and we can close this and I will instead maintain this for our internal use out of tree for now.

Thanks.

Reply:
Hi @mcscholtz ,

I was looking others SoC to get a better view. No objections to move forward with this change. I only recommend a better description for `reg-rc`, see example below:

_Register C compare/match value. RC can be used as compare/match unit for an specific timer unit. Their use depends on how timer channel is configured, see reg-cmr. It can be used as trigger for both input capture or counter mode, or even as event source. The RC register behaves is SoC dependent. For more information and use cases, check SoC datasheet and application notes._

Reply:
@nandojve  Great thanks, I have updated the PR with your suggested description.